pkg/eth/api.go

```diff
@@ -698,7 +698,7 @@ func (pea *PublicEthAPI) GetStorageAt(ctx context.Context, address common.Addres
 		}
 	}
 	if err == sql.ErrNoRows {
-		return make([]byte, 32), nil
+		return hexutil.Bytes{}, errors.New("header not found")
```

Comment:
Did you confirm this behaviour against the standard JSON RPC API endpoint, i.e. expected result when the storage slot doesn't exist.

Reply:
Yeah I think this is undoing one of the fixes Ramil made here, the unit test is broken in what it is expecting rather than what it is receiving in this case (and probably all the cases where the unit tests were broken by the changes made in this PR, as the unit tests were passing before them). The majority of the problems this PR is fixing at a high level are due to me not anticipating (and failing to test) what the correct expected results were for error and nil cases.

Reply:
Yes, I confirmed the behavior manually. 
We have two tests suite:
**Unit test:** Which makes calls to the DB and checks expectations.
**Integration test**: Which compares ETH and IPLD APIs response.
I will check if there is an Integration test for the same or else I will add it.